toolchain/check/context.h

```diff
@@ -384,6 +389,10 @@ class Context {
   // Inline 0 elements because it's expected to require heap allocation.
   llvm::SmallVector<SemIR::ConstantValueStore, 0> import_ir_constant_values_;
 
+  // Per-Carbon-file start locations for corresponding Clang source buffers.
+  // Owned and managed by code in cpp/location.cpp.
+  llvm::SmallVector<clang::SourceLocation, 0> cpp_carbon_file_locations_;
```

Comment:
Is the `, 0` going to be useful here? Often we're going to have a smaller number of files (or errors might be in the current file, or the API -- lower indices), maybe it'd actually fit inline for `SmallVector`?

```suggestion
  llvm::SmallVector<clang::SourceLocation> cpp_carbon_file_locations_;
```